container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java

```diff
@@ -109,6 +112,35 @@
 
     protected abstract void doPartialFill(Result result, String summaryClass);
 
+    protected static boolean wantsRPCSummarFill(Query query) {
+        return query.properties().getBoolean(dispatchSummaries);
+    }
+
+    /**
+     * Returns whether we need to send the query when fetching summaries.
+     * This is necessary if the query requests summary features or dynamic snippeting
+     */
+    protected boolean summaryNeedsQuery(Query query) {
+        if (query.getRanking().getQueryCache()) return false;  // Query is cached in backend
+
+        DocumentDatabase documentDb = getDocumentDatabase(query);
+
+        // Needed to generate a dynamic summary?
+        DocsumDefinition docsumDefinition = documentDb.getDocsumDefinitionSet().getDocsumDefinition(query.getPresentation().getSummary());
+        if (docsumDefinition == null) return true; // stay safe
+        if (docsumDefinition.isDynamic()) return true;
+
+        // Needed to generate ranking features?
+        RankProfile rankProfile = documentDb.rankProfiles().get(query.getRanking().getProfile());
+        if (rankProfile == null) return true; // stay safe
+        if (rankProfile.hasSummaryFeatures()) return true;
+        if (query.getRanking().getListFeatures()) return true;
+
+        // (Don't just add other checks here as there is a return false above)
```

Comment:
That could be said about any function. Also, I think the return false earlier implies less thinking, not more. I was just curious about exactly which checks the author was afraid someone would add.